velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -291,38 +287,173 @@ TpchPlan TpchQueryBuilder::getQ6Plan() const {
                   .planNode();
   TpchPlan context;
   context.plan = std::move(plan);
-  context.dataFiles[lineitemPlanNodeId] = getTableFilePaths(kLineitem);
+  context.dataFiles[lineitemPlanNodeId] = getTableFilePaths(kLineitem_);
+  context.dataFileFormat = format_;
+  return context;
+}
+
+TpchPlan TpchQueryBuilder::getQ10Plan() const {
+  std::vector<std::string> customerColumns = {
+      "c_nationkey",
+      "c_custkey",
+      "c_acctbal",
+      "c_name",
+      "c_address",
+      "c_phone",
+      "c_comment"};
+  std::vector<std::string> nationColumns = {"n_nationkey", "n_name"};
+  std::vector<std::string> lineitemColumns = {
+      "l_orderkey", "l_returnflag", "l_extendedprice", "l_discount"};
+  std::vector<std::string> ordersColumns = {
+      "o_orderdate", "o_orderkey", "o_custkey"};
+
+  auto customerSelectedRowType = getRowType(kCustomer_, customerColumns);
+  const auto& customerFileColumns = getFileColumnNames(kCustomer_);
+  auto nationSelectedRowType = getRowType(kNation_, nationColumns);
+  const auto& nationFileColumns = getFileColumnNames(kNation_);
+  auto lineitemSelectedRowType = getRowType(kLineitem_, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem_);
+  auto ordersSelectedRowType = getRowType(kOrders_, ordersColumns);
+  const auto& ordersFileColumns = getFileColumnNames(kOrders_);
+
+  std::string lineitemReturnFlagFilter = "l_returnflag = 'R'";
+  const auto orderDate = "o_orderdate";
+  std::string orderDateFilter;
+  // DWRF does not support Date type. Use Varchar instead.
```

Comment:
This code repeats a lot. Is there a way to refactor to avoid copy-paste?

Reply:
Thanks for the suggestion, added procedure 'formatDateFilter' to address this.